pkg/placement/placement-resolver.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// PlacementResolver abstracts the functionality needed from a placement resolver,
+// to simplify its usage and keep implementation details inaccessible.
+type PlacementResolver interface {
+	// GeneratePlacementDecision returns the placement decision for the given
+	// placement key. If a key does not exist in the cache, it is created.
+	GeneratePlacementDecision(placementKey string) (*v1alpha1.PlacementDecisionSpec, error)
```

Comment:
I think the reader should be told something about what errors might arise here. In a controller it is important to know whether an error is indicating a problem that might be cured by a retry later or cannot be cured by a retry later.

If I understand correctly, the only error that can arise here is from a missing entry in a `util.GvkGvrMapper`. Since there is a lot of asynchrony going on here, that is something that actually can happen and is not the result of any user or code doing anything wrong; it _is_ something that is likely to be cured by a retry later.